diff --git a/src/Makefile.am b/src/Makefile.am
index 18d1244b..e6e56b12 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -35,7 +35,7 @@ BASE_SOURCES=main.c rtp.h rtpp_server.c rtpp_server.h \
   rtpp_netaddr.c rtpp_netaddr.h rtpp_acct_pipe.h
 
 rtpproxy_LDADD=-lm -lpthread
-rtpproxy_debug_LDADD=${rtpproxy_LDADD} $(top_srcdir)/libexecinfo/libexecinfo.a \
+rtpproxy_debug_LDADD=${rtpproxy_LDADD} $(top_builddir)/libexecinfo/libexecinfo.a \
   @LIBS_DL@
 rtpproxy_debug_LDFLAGS=-static -Wl,--export-dynamic
 
@@ -77,92 +77,92 @@ GENFINCODE= $(top_srcdir)/src/genfincode.sh
 GENFINCODE_S= $(top_srcdir)/src/genfincode_stat.sh
 
 rtpp_timed_fin.c: genfincode.sh rtpp_timed.h
-	$(GENFINCODE) rtpp_timed.h rtpp_timed_fin.h rtpp_timed_fin.c
+	$(GENFINCODE) $(top_srcdir)/src/rtpp_timed.h rtpp_timed_fin.h rtpp_timed_fin.c
 
 rtpp_timed_fin.h: rtpp_timed_fin.c
 
 rtpp_stream_fin.c: $(GENFINCODE_S) rtpp_stream.h
-	$(GENFINCODE_S) rtpp_stream.h rtpp_stream_fin.h rtpp_stream_fin.c
+	$(GENFINCODE_S) $(top_srcdir)/src/rtpp_stream.h rtpp_stream_fin.h rtpp_stream_fin.c
 
 rtpp_stream_fin.h: rtpp_stream_fin.c
 
 rtpp_server_fin.c: $(GENFINCODE) rtpp_server.h
-	$(GENFINCODE_S) rtpp_server.h rtpp_server_fin.h rtpp_server_fin.c
+	$(GENFINCODE_S) $(top_srcdir)/src/rtpp_server.h rtpp_server_fin.h rtpp_server_fin.c
 
 rtpp_server_fin.h: rtpp_server_fin.c
 
 rtpp_refcnt_fin.c: $(GENFINCODE_S) rtpp_refcnt.h
-	$(GENFINCODE_S) rtpp_refcnt.h rtpp_refcnt_fin.h rtpp_refcnt_fin.c
+	$(GENFINCODE_S) $(top_srcdir)/src/rtpp_refcnt.h rtpp_refcnt_fin.h rtpp_refcnt_fin.c
 
 rtpp_refcnt_fin.h: rtpp_refcnt_fin.c
 
 rtpp_log_obj_fin.c: $(GENFINCODE) rtpp_log_obj.h
-	$(GENFINCODE) rtpp_log_obj.h rtpp_log_obj_fin.h rtpp_log_obj_fin.c
+	$(GENFINCODE) $(top_srcdir)/src/rtpp_log_obj.h rtpp_log_obj_fin.h rtpp_log_obj_fin.c
 
 rtpp_log_obj_fin.h: rtpp_log_obj_fin.c
 
 rtpp_command_rcache_fin.c: $(GENFINCODE) rtpp_command_rcache.h
-	$(GENFINCODE) rtpp_command_rcache.h rtpp_command_rcache_fin.h rtpp_command_rcache_fin.c
+	$(GENFINCODE) $(top_srcdir)/src/rtpp_command_rcache.h rtpp_command_rcache_fin.h rtpp_command_rcache_fin.c
 
 rtpp_command_rcache_fin.h: rtpp_command_rcache_fin.c
 
 rtpp_socket_fin.c: $(GENFINCODE) rtpp_socket.h
-	$(GENFINCODE) rtpp_socket.h rtpp_socket_fin.h rtpp_socket_fin.c
+	$(GENFINCODE) $(top_srcdir)/src/rtpp_socket.h rtpp_socket_fin.h rtpp_socket_fin.c
 
 rtpp_socket_fin.h: rtpp_socket_fin.c
 
 rtpp_record_fin.c: $(GENFINCODE) rtpp_record.h
-	$(GENFINCODE) rtpp_record.h rtpp_record_fin.h rtpp_record_fin.c
+	$(GENFINCODE) $(top_srcdir)/src/rtpp_record.h rtpp_record_fin.h rtpp_record_fin.c
 
 rtpp_record_fin.h: rtpp_record_fin.c
 
 rtpp_ttl_fin.c: $(GENFINCODE) rtpp_ttl.h
-	$(GENFINCODE) rtpp_ttl.h rtpp_ttl_fin.h rtpp_ttl_fin.c
+	$(GENFINCODE) $(top_srcdir)/src/rtpp_ttl.h rtpp_ttl_fin.h rtpp_ttl_fin.c
 
 rtpp_ttl_fin.h: rtpp_ttl_fin.c
 
 rtpp_pipe_fin.c: $(GENFINCODE) rtpp_pipe.h
-	$(GENFINCODE) rtpp_pipe.h rtpp_pipe_fin.h rtpp_pipe_fin.c
+	$(GENFINCODE) $(top_srcdir)/src/rtpp_pipe.h rtpp_pipe_fin.h rtpp_pipe_fin.c
 
 rtpp_pipe_fin.h: rtpp_pipe_fin.c
 
 rtpp_pcount_fin.c: $(GENFINCODE) rtpp_pcount.h
-	$(GENFINCODE) rtpp_pcount.h rtpp_pcount_fin.h rtpp_pcount_fin.c
+	$(GENFINCODE) $(top_srcdir)/src/rtpp_pcount.h rtpp_pcount_fin.h rtpp_pcount_fin.c
 
 rtpp_pcount_fin.h: rtpp_pcount_fin.c
 
 rtpp_ringbuf_fin.c: $(GENFINCODE) rtpp_ringbuf.h
-	$(GENFINCODE) rtpp_ringbuf.h rtpp_ringbuf_fin.h rtpp_ringbuf_fin.c
+	$(GENFINCODE) $(top_srcdir)/src/rtpp_ringbuf.h rtpp_ringbuf_fin.h rtpp_ringbuf_fin.c
 
 rtpp_ringbuf_fin.h: rtpp_ringbuf_fin.c
 
 rtpp_sessinfo_fin.c: $(GENFINCODE) rtpp_sessinfo.h
-	$(GENFINCODE) rtpp_sessinfo.h rtpp_sessinfo_fin.h rtpp_sessinfo_fin.c
+	$(GENFINCODE) $(top_srcdir)/src/rtpp_sessinfo.h rtpp_sessinfo_fin.h rtpp_sessinfo_fin.c
 
 rtpp_sessinfo_fin.h: rtpp_sessinfo_fin.c
 
 rtpp_pcnt_strm_fin.c: $(GENFINCODE) rtpp_pcnt_strm.h
-	$(GENFINCODE) rtpp_pcnt_strm.h rtpp_pcnt_strm_fin.h rtpp_pcnt_strm_fin.c
+	$(GENFINCODE) $(top_srcdir)/src/rtpp_pcnt_strm.h rtpp_pcnt_strm_fin.h rtpp_pcnt_strm_fin.c
 
 rtpp_pcnt_strm_fin.h: rtpp_pcnt_strm_fin.c
 
 rtpp_module_if_fin.c: $(GENFINCODE) rtpp_module_if.h
-	$(GENFINCODE) rtpp_module_if.h rtpp_module_if_fin.h rtpp_module_if_fin.c
+	$(GENFINCODE) $(top_srcdir)/src/rtpp_module_if.h rtpp_module_if_fin.h rtpp_module_if_fin.c
 
 rtpp_module_if_fin.h: rtpp_module_if_fin.c
 
 rtpp_port_table_fin.c: $(GENFINCODE) rtpp_port_table.h
-	$(GENFINCODE) rtpp_port_table.h rtpp_port_table_fin.h rtpp_port_table_fin.c
+	$(GENFINCODE) $(top_srcdir)/src/rtpp_port_table.h rtpp_port_table_fin.h rtpp_port_table_fin.c
 
 rtpp_port_table_fin.h: rtpp_port_table_fin.c
 
 rtpp_acct_fin.c: $(GENFINCODE) rtpp_acct.h
-	$(GENFINCODE) rtpp_acct.h rtpp_acct_fin.h rtpp_acct_fin.c
+	$(GENFINCODE) $(top_srcdir)/src/rtpp_acct.h rtpp_acct_fin.h rtpp_acct_fin.c
 
 rtpp_acct_fin.h: rtpp_acct_fin.c
 
 rtpp_netaddr_fin.c: $(GENFINCODE_S) rtpp_netaddr.h
-	$(GENFINCODE_S) rtpp_netaddr.h rtpp_netaddr_fin.h rtpp_netaddr_fin.c
+	$(GENFINCODE_S) $(top_srcdir)/src/rtpp_netaddr.h rtpp_netaddr_fin.h rtpp_netaddr_fin.c
 
 rtpp_netaddr_fin.h: rtpp_netaddr_fin.c
 
diff --git a/src/genfincode.sh b/src/genfincode.sh
index d5eefa8c..14b6f039 100755
--- a/src/genfincode.sh
+++ b/src/genfincode.sh
@@ -12,7 +12,7 @@ gen_fin_c() {
   echo "#include <stdlib.h>"
   echo "#include \"rtpp_types.h\""
   echo "#include \"rtpp_debug.h\""
-  DEFNAME=`echo ${1} | sed 's|[.]|_|g'`
+  DEFNAME=`echo $(basename ${1}) | sed 's|[.]|_|g'`
   echo "#define ${DEFNAME}_fin 1"
   echo "#include \"${1}\""
 
diff --git a/makeann/Makefile.am b/makeann/Makefile.am
index 7780021d..0a936750 100644
--- a/makeann/Makefile.am
+++ b/makeann/Makefile.am
@@ -6,5 +6,6 @@ makeann_SOURCES=makeann.c rtp.h g711.h ${extractaudio_SRCDIR}/g729_compat.c \
   ${extractaudio_SRCDIR}/g729_compat.h
 makeann_LDADD=@LIBS_G729@ @LIBS_GSM@ @LIBS_G722@
 makeann_CPPFLAGS=-I${extractaudio_SRCDIR}
+INCLUDES=-I$(top_srcdir)/src
 
 DEFS=	-Wall @DEFS@

diff --git a/modules/acct_csv/Makefile.am b/modules/acct_csv/Makefile.am
index f0e128c2..91753385 100644
--- a/modules/acct_csv/Makefile.am
+++ b/modules/acct_csv/Makefile.am
@@ -4,8 +4,9 @@ rtpp_acct_csv_la_SOURCES = rtpp_acct_csv.c
 rtpp_acct_csv_la_LDFLAGS = -avoid-version -module -shared -export-dynamic
 
 rtpp_acct_csv_debug_la_SOURCES = ${rtpp_acct_csv_la_SOURCES}
-rtpp_acct_csv_debug_la_LIBADD=$(top_srcdir)/libexecinfo/libexecinfo.a \
+rtpp_acct_csv_debug_la_LIBADD=$(top_builddir)/libexecinfo/libexecinfo.a \
   @LIBS_DL@
 rtpp_acct_csv_debug_la_LDFLAGS=${rtpp_acct_csv_la_LDFLAGS}
 rtpp_acct_csv_debug_la_CPPFLAGS=-DRTPP_DEBUG
 rtpp_acct_csv_debug_la_CFLAGS=-g3 -O0 -fno-omit-frame-pointer
+INCLUDES = -I$(top_srcdir)/src

diff --git a/extractaudio/Makefile.am b/extractaudio/Makefile.am
index a97c914f..002bd96b 100644
--- a/extractaudio/Makefile.am
+++ b/extractaudio/Makefile.am
@@ -1,5 +1,6 @@
 bin_PROGRAMS=extractaudio
-MAINSRCDIR=$(top_srcdir)/src
+INCLUDES=-I$(top_srcdir)/src
+MAINSRCDIR=$(top_builddir)/src
 extractaudio_SOURCES=main.c decoder.c rtpp_loader.c ${MAINSRCDIR}/rtp.c \
   ${MAINSRCDIR}/rtp_analyze.c ${MAINSRCDIR}/rtpp_util.c ${MAINSRCDIR}/rtpp_time.c \
   ${MAINSRCDIR}/rtpp_log_stand.c decoder.h format_au.h g711.h rtpp_loader.h \
